repository/src/main/java/org/apache/atlas/repository/store/graph/v2/EntityGraphMapper.java

```diff
@@ -2928,6 +2932,10 @@ private AtlasEdge updateRelationship(AtlasEdge currentEdge, final AtlasVertex pa
         String    newEntityId = getIdFromVertex(newEntityVertex);
         AtlasEdge ret         = currentEdge;
 
+        if(StringUtils.isEmpty(currentEntityId)){
```

Comment:
I fear skipping empty value for currentEntityId might lead any unknown issue
Can we only check for `currentEntityId == null` here?